libmscore/score.cpp

```diff
@@ -1978,148 +1978,182 @@ void Score::removeAudio()
+                  }
+            // check if a clef change is present but is spurious (i.e. no actual change)
+            else if (staff->currentClefTick(ctick) == ctick &&
+                     staff->clef(ctick - Fraction::fromTicks(1)) == ostaff->clef(otick)) {
+                        Segment* ns = firstAppendedMeasure->first(SegmentType::Clef);
+                        if (!ns)
+                              ns = firstAppendedMeasure->first(SegmentType::HeaderClef);
+                        if (ns)
+                              ns->remove(ns->element(trackIdx));
                   }
             }
 
-      // clone the spanners
-      for (auto sp : score->spanner()) {
-            Spanner* spanner = sp.second;
+      // check if section starts with a pick-up measure to be merged with end of previous section
+      Measure* cm = firstAppendedMeasure, * pm = cm->prevMeasure();
+      if (pm->timesig() == cm->timesig() && pm->ticks() + cm->ticks() == cm->timesig())
+            cmdJoinMeasure(pm, cm);
+
+      // clone the spanners (only in the range currently copied)
+      auto ospans = score->spanner();
+      auto lb = ospans.lower_bound(startTick.ticks()), ub = ospans.upper_bound(endTick.ticks());
+      for (auto sp = lb; sp != ub; sp++) {
+            Spanner* spanner = sp->second;
+            
+            if (spanner->tick2() > endTick) continue; // map is by tick() so this can still happen in theory...
+
             Spanner* ns = toSpanner(spanner->clone());
             ns->setScore(this);
             ns->setParent(0);
-            ns->setTick(spanner->tick() + tickOfAppend);
-            ns->setTick2(spanner->tick2() + tickOfAppend);
-            if (ns->type() == ElementType::SLUR) {
-                  // set start/end element for slur
-                  ns->setStartElement(0);
-                  ns->setEndElement(0);
-                  Measure* sm = tick2measure(ns->tick());
-                  if (sm)
-                        ns->setStartElement(sm->findChordRest(ns->tick(), ns->track()));
-                  Measure * em = tick2measure(ns->tick2());
-                  if (em)
-                        ns->setEndElement(em->findChordRest(ns->tick2(), ns->track2()));
-                  if (!ns->startElement())
-                        qDebug("clone Slur: no start element");
-                  if (!ns->endElement())
-                        qDebug("clone Slur: no end element");
-                  }
+            ns->setTick(spanner->tick() - startTick + tickOfAppend);
+            ns->setTick2(spanner->tick2() - startTick + tickOfAppend);
+            ns->computeStartElement();
+            ns->computeEndElement();
             addElement(ns);
             }
+
+      return true;
+      }
+
+//---------------------------------------------------------
+//   appendScore
+//---------------------------------------------------------
+
+bool Score::appendScore(Score* score, bool addPageBreak, bool addSectionBreak)
```

Comment:
Could this function perhaps be moved above the `appendMeasuresFromScore`? It looks like it remained unchanged after this PR (if I am correct), and the version control system doesn't understand that. If it is possible, it would be better if git could handle this correctly.